packages/nextjs/components/ExampleUi/ContractInteraction.tsx

```diff
@@ -3,14 +3,18 @@ import { DiamondIcon } from "./assets/DiamondIcon";
 import { HareIcon } from "./assets/HareIcon";
 import { ArrowSmallRightIcon, XMarkIcon } from "@heroicons/react/24/outline";
 import { useState } from "react";
-import { useScaffoldContractWrite } from "~~/hooks/scaffold-eth";
+import { useScaffoldContractWrite, useScaffoldContractEvent } from "~~/hooks/scaffold-eth";
```

Comment:
Minor thing but: maybe we could move this to `ContractData.tsx`, since it's a read op?

Reply:
hmm, why do we need it? Maybe remove it for now?

Reply:
> Minor thing but: maybe we could move this to ContractData.tsx, since it's a read op?

💯, moved it at [25cda29](https://github.com/scaffold-eth/se-2/pull/162/commits/25cda2943095ac547fe0ecbc32cbed1ec9e0737b)

> hmm, why do we need it? Maybe remove it for now?
Yup makes sense too

Reply:
Listening for all the events will be very tricky except if we build a mini-event listener and make use of other packages or since the graph will still be added to this build, we should just make it simpler and easy to use for anyone making use of it.

Reply:
> Listening for all the events will be very tricky except if we build a mini-event listener and make use of other packages or since the graph will still be added to this build, we should just make it simpler and easy to use for anyone making use of it.

SE-1 (with eth-hooks) does it. You can specify an start block when calling the `useEventListener` hook.